x-pack/auditbeat/module/system/process/_meta/data.json

```diff
@@ -0,0 +1,34 @@
+{
+    "@timestamp": "2017-10-12T08:05:34.853Z",
+    "agent": {
+        "hostname": "host.example.com",
+        "name": "host.example.com"
+    },
+    "event": {
+        "action": "existing_process",
+        "dataset": "process",
+        "id": "74556762-5bba-4e9a-abac-25c283a78da8",
+        "module": "system",
+        "type": "state"
+    },
+    "process": {
+        "name": "systemd",
+        "pid": 1
+    },
+    "system": {
+        "process": {
+            "args": [
```

Comment:
`args` and `ppid` can go under `^process.` ([ecs ref](https://github.com/elastic/ecs#-process-fields)).

Reply:
I thought about adding those top-level. I didn't for two reasons:

1. `args` can become quite big. With a process like Elasticsearch that has many (and long) arguments storing `args` twice can nearly double the document size.
2. To me, the main purpose of adding fields to the top-level is to allow quick joins between documents from different data sources - e.g. joining a socket to a process to a user to a login event all just using the common field `user.id` (or in this case a socket to a process using `process.pid`). I don't think `args` would be used for that purpose. `ppid` maybe.

Reply:
I think it's hard to know exactly how and why someone will query this information. Rather than doing a join, you might be doing an initial search where you want to find any processes that used a certain argument (like a file name). So by populating `process.args` you will enable them to find both system/process events and auditd events thanks to ECS.

So I'd rather see this populating the `process` namespace as much as possible and omitting that same information from the `system.process` namespace. If necessary we could utilize an alias field like system.process.args -> process.args, but I'd wait to see if that's something users need.

Reply:
Could we just pull all `system.process` fields to `process`? ECS does have most of them already, except `cwd`, `exe`, and `starttime` - all of which I could see being added to ECS, if not now then later.

Reply:
I think all of those should be part of the `process` in ECS. There's already a [PR open](https://github.com/elastic/ecs/pull/209) to add `process.executable`.

I'd probably go with `process.start` to be consistent with `event.start`.

Reply:
I agree with Andrew's point about leveraging `process.` as much as possible. You're free to keep a copy or alias to ECS field in your fieldset if you want.

I made a note of adding `process.working_directory` and `process.start_time` to ECS. I agree they will end up in there at some point, this is very basic stuff. I'd say don't depend on it just yet, but it's likely to make beta 2, since they're non-controversial fields and it's a pretty straightforward change.